### ISSUE TITLE
Hipchat notifications

### ISSUE THREAD

**Issue by REBELinBLUE** [REBELinBLUE/deployer#159]
Along with slack it might be nice having hipchat

**Comment by lorvent** [comment]
Yes,
its always good to have hipchat notification aswell as we use hipchat in our organization.

**Comment by lorvent** [comment]
does this help?

https://github.com/hernandev/hipchat-laravel

**Comment by REBELinBLUE** [comment]
Possibly, I'll try and take a look tonight after I finish #165 but I'd imagine it is fairly simple, something like the slack one https://github.com/REBELinBLUE/deployer/blob/master/app/Jobs/SlackNotify.php#L40-L61

Doesn't look like a lot of work from looking at PHPCI https://github.com/Block8/PHPCI/blob/master/PHPCI/Plugin/HipchatNotify.php#L74-L93 but I note that Atlassian has changed the API and are requesting people use the new version

**Comment by lorvent** [comment]
Good Morning :) 
yeah, it looks pretty easy and since you already implemented slack, it should not be tough at all.

**Comment by REBELinBLUE** [comment]
Looks relatively simple, basically the same as Slack

````php
Request::post($url)
              ->sendsJson()
              ->body([
                   'from' => 'deployer', 
                   'color'   => 'red', // green,
                   'message' => '<b>failed!</b>',
                   'message_format' => 'html', // text
              ])->send();
````

https://www.hipchat.com/docs/apiv2/method/send_room_notification

May try and get this done tomorrow as it is a quick win

**Comment by lorvent** [comment]
wow.... I'm excited :heart_eyes:

**Comment by lorvent** [comment]
Hello,

looks like you made lot of progress but not included that branch into master,
any reason behind that?

thanks.

**Comment by REBELinBLUE** [comment]
It isn't finished yet, unfortunately I have been busy the last week

**Comment by lorvent** [comment]
Oh.. ok

Thanks

**Comment by lorvent** [comment]
Hi,

you can use monolog handler aswell for a quick way of doing it

https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/HipChatHandler.php

something like https://github.com/tylercd100/lern/blob/master/src/Factories/MonologHandlerFactory.php#L42

**Comment by REBELinBLUE** [comment]
https://github.com/laravel-notification-channels should make this much simpler

**Comment by lorvent** [comment]
I am thinking same as soon as they announce 5.3 at laracon

but there is no direct support for hipchat :cry:

**Comment by REBELinBLUE** [comment]
Nope, but there people have already done it http://laravel-notification-channels.com/ run by the guys from @spatie who provide loads of Laravel packages 😃

**Comment by lorvent** [comment]
Damn.... so fast :smile: 

i saw that package is trending but never thought they added it so quickly.

When you will be upgrading it to 5.3 :stuck_out_tongue:

**Comment by REBELinBLUE** [comment]
5.3 isn't out yet ;) But someone has backported it http://laravel-notification-channels.com/backport/ I was going to have a play around at the weekend if I get a chance :)

**Comment by lorvent** [comment]
Taylor said he will be releasing at laracon EU so better to wait for it.

**Comment by REBELinBLUE** [comment]
True but I can at least start playing with it, I'm sure there are other useful features is 5.3

**Comment by REBELinBLUE** [comment]
Solved by #297
